src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java

```diff
@@ -59,13 +63,15 @@
     private final CustomLocalDragboard localDragBoard;
     private final ObservableList<BibEntry> entriesList;
     private final DelayTaskThrottler throttler;
+    @Inject private PreferencesService preferencesService;
 
     public GroupNodeViewModel(BibDatabaseContext databaseContext, StateManager stateManager, TaskExecutor taskExecutor, GroupTreeNode groupNode, CustomLocalDragboard localDragBoard) {
         this.databaseContext = Objects.requireNonNull(databaseContext);
         this.taskExecutor = Objects.requireNonNull(taskExecutor);
         this.stateManager = Objects.requireNonNull(stateManager);
         this.groupNode = Objects.requireNonNull(groupNode);
         this.localDragBoard = Objects.requireNonNull(localDragBoard);
+        this.preferencesService = JabRefPreferences.getInstance();
```

Comment:
This should not be necessary, as you already inject the PreferencesService in line 66.
However, I think you ran into a similar bug as I did in #6107 . The dependency injection is somewhat broken, as it probably does not inject the preferencesService in a viewmodel. As a work around, you could inject the PreferencesService by the constructor call. This should work until the other problem is fixed.

Reply:
Hi @calixtus ,
thanks for your comment. I'll check issue #6107 and try to inject PreferencesService by the constructor call.
Thanks for reminder about checkboxes. I'm a new to this project (and Java) and need a bit of guidance so far ))))

Reply:
Welcome to JabRef. If you have questions, we primarily us the JabRef-gitter-channel for development discussion. Don't hesitate to contact us.

Reply:
@Gena928 Did you already find the time to have a look at the injection business?

Reply:
Hi @tobiasdiez,
I was able to resolve the problem using `GroupNodeViewModel.java` properties:
- removed call to `calculateNumberOfMatches()` method from constructor;
- added private field and method for calling it from outside;
- called it from `GroupTreeView.java`;
Works.

As for constructor call, sorry, can't understand 😩@calixtus kindly explained that to me in gitter:
Just think of `.withGraphic` as a method: 
```
class ViewModelCell { 
 void withGraphic(GroupNodeViewModel group) { /* do something with the group */ } 
}
```
But I still can't get, how can I pass a Pref file using that method.

Reply:
Create a constructor argument for PreferencesService in GroupNodeViewModel.
Now you need to find where the constructor of GroupNodeViewModel is called (will be clear) and there you pass the PreferencesService again.

Reply:
You don't need to modfiy the ViewModelCell stuff, it's just for displaying in the component